src/core/resource.cpp

```diff
@@ -267,6 +269,11 @@ EMGMResponse CResource::writeValue(forte::core::TNameIdentifier &paNameList, con
             //if we have got a connection it was a DO mirror the forced value there
             con->writeData(*var);
           }
+        }else{
+          InitValue value;
+          value.ptr = var;
+          value.str = paValue.getStorage().c_str();
+          mInitList.push_back(value);
```

Comment:
Why are you using `c_str()` when `getStorage()` already gives you a `std::string`? I would also suggest avoiding the intermediate local variable by using [emplace_back()](https://en.cppreference.com/w/cpp/container/vector/emplace_back) to construct the `InitValue` in-place.